src/providers/xref.provider.ts

```diff
@@ -30,15 +30,41 @@ function shouldProvide (context: Context): boolean {
 }
 
 async function getLabels (): Promise<string[]> {
-  const regex = /\\[\\[(\\w+)\\]\\]/g
-  const labels = await vscode.workspace.findFiles('**/*.adoc').then((files) =>
-    files
-      .map((uri) => readFileSync(uri.path).toString('utf-8'))
-      .join('\n')
-      .match(regex)
-      .map((result) => result.replace('[[', '').replace(']]', ''))
-  )
-  return labels
+  const files = await vscode.workspace.findFiles('**/*.adoc')
+  const contentOfFilesConcatenated = files
+    .map((uri) => readFileSync(uri.path).toString('utf-8'))
+    .join('\n')
+  const labelsFromLegacyBlock = await getLabelsFromLegacyBlock(contentOfFilesConcatenated)
+  const labelsFromShorthandNotation = await getLabelsFromShorthandNotation(contentOfFilesConcatenated)
+  const labelsFromLonghandNotation = await getLabelsFromLonghandNotation(contentOfFilesConcatenated)
+  return labelsFromLegacyBlock.concat(labelsFromShorthandNotation, labelsFromLonghandNotation)
+}
+
+async function getLabelsFromLonghandNotation (content: string): Promise<string[]> {
+  const regex = /\[id=(\w+)\]/g
+  const matched = content.match(regex)
+  if (matched) {
+    return matched.map((result) => result.replace('[id=', '').replace(']', ''))
```

Comment:
we should probably ignore everything after the `]` or make sure that the line ends with `]`

Reply:
right, it is necessary to support inlined anchor https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#as-an-inline-anchor

Reply:
also for additional anchors to a section https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#add-additional-anchors-to-a-section

Reply:
in fact, it is already working because ti is using the matched string, so there is nothign after the `]` I added a test case.

![image](https://user-images.githubusercontent.com/1105127/203994572-93ba93cb-2c06-4ff1-a154-b7d92fed35a8.png)